### PR TITLE
Added Devices tab in Settings to display list of connected devices

### LINKED ISSUE #1222: Display list of connected devices on the Web Client
**Actual Behaviour**

There is no section for Devices on the web client.

**Expected Behaviour**

The devices which are linked to the logged in user should be fetched from the APIs created in [susi_server #740](https://github.com/fossasia/susi_server/pull/740) and then be displayed on the web client.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by Akshat-Jain**
Fixes #1222 

Changes: Added Devices tab in Settings to display list of connected devices

Demo Link: https://pr-1222-fossasia-susi-web-chat.surge.sh/

Screenshots for the change: 
devicestabinsettings

### CODE REVIEW

**Comment by Akshat-Jain** [comment]
@PrP-11 @akshatnitd @anshumanv @DNS-404 @imjog Please review. Thanks :-)

File: src/components/MyDevices/MyDevices.react.js [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+import Translate from '../Translate/Translate.react';
+import TableComplex from '../TableComplex/TableComplex.react';
+
+import { Link } from 'react-router-dom';
+
+import './MyDevices.css'
+
+const cookies = new Cookies();
+let obj=[];
+let url = 'https://api.susi.ai/aaa/listUserSettings.json?'
```
Comment (akshatnitd) [inline src/components/MyDevices/MyDevices.react.js]: Do not hardcode. Use `cookies.get('serverUrl')` for server URL

File: src/components/MyDevices/MyDevices.css [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+.section {
+  padding: 0px 20px 20px 20px !important;
+}
+
+.table {
+  position: relative !important;
+  left: 32%;
+  margin-top: 20px;
+}
```
Comment (akshatnitd) [inline src/components/MyDevices/MyDevices.css]: Add EOF

File: src/components/ChatApp/TopBar.react.js [hunk 3]
```diff
@@ -101,11 +102,15 @@ class TopBar extends Component {
...
-					<MenuItem
-						rightIcon={<Extension/>}
-						href="https://skills.susi.ai/botbuilder"
+						<MenuItem
+							rightIcon={<Extension/>}
+							href="https://skills.susi.ai/botbuilder"
 					><Translate text="Botbuilder"/>
 					</MenuItem>
+					<MenuItem
+                    containerElement={<Link to="/mydevices" />}
```
Comment (akshatnitd) [inline src/components/ChatApp/TopBar.react.js]: Indentation issues

File: src/components/MyDevices/MyDevices.react.js (hunk 1 above)
Comment (akshatnitd) [reply src/components/MyDevices/MyDevices.react.js]: Try to break the URL into BASE_URL + route

Comment (akshatnitd) [reply src/components/MyDevices/MyDevices.react.js]: Refer API.actions.js

File: src/components/MyDevices/MyDevices.react.js [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+          503: function(xhr) {
+            if(window.console) {console.log(xhr.responseText);}
+            console.log('Error 503: Server not responding!');
+          }
+        },
+        crossDomain: true,
+        timeout: 3000,
+        async: false,
+        success: function (response) {
+
```
Comment (akshatnitd) [inline src/components/MyDevices/MyDevices.react.js]: Remove unnecessary blank lines

**Comment by Akshat-Jain** [comment]
@akshatnitd I've made the suggested changes. Please review. Thanks :-)

File: src/components/MyDevices/MyDevices.react.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            if(window.console) {console.log(xhr.responseText);}
+            console.log('Error 503: Server not responding!');
+          }
+        },
+        crossDomain: true,
+        timeout: 3000,
+        async: false,
+        success: function (response) {
+            let keys = Object.keys(response.devices);
+            for (var i = 0; i < keys.length; i++) {
```
Comment (anshumanv) [inline src/components/MyDevices/MyDevices.react.js]: Nobody uses this now, you can use something like `for .. of ..` or `forEach()..`

File: src/components/MyDevices/MyDevices.react.js [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            }
+        }.bind(this),
+        error: function(errorThrown){
+          console.log(errorThrown);
+        }
+      });
+    }
+
+    // eslint-disable-next-line
+    componentWillMount(){
```
Comment (anshumanv) [inline src/components/MyDevices/MyDevices.react.js]: Why not cDM?

File: src/components/MyDevices/MyDevices.react.js (hunk 5 above)
Comment (Akshat-Jain) [reply src/components/MyDevices/MyDevices.react.js]: Okay, I will make the changes. Thanks :-)

File: src/components/MyDevices/MyDevices.react.js (hunk 6 above)
Comment (Akshat-Jain) [reply src/components/MyDevices/MyDevices.react.js]: cDM was resulting in mounting/unmounting issues, that's why.

Comment (anshumanv) [reply src/components/MyDevices/MyDevices.react.js]: Which issues exactly, I think it should work fine. :thinking:

**Comment by Akshat-Jain** [comment]
@akshatnitd @anshumanv Did all the changes as requested by you. Please review. Thanks :-)

**Comment by Akshat-Jain** [comment]
@isuruAb @madhavrathi Please review. Thanks :-)

**Comment by DNS-404** [comment]
It's behaving strangely if we use custom themes. Please have a look at it. Rest looks fine 👍 

screen shot 2018-05-27 at 3 40 44 pm

File: src/components/MyDevices/MyDevices.react.js [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+          }
+        },
+        crossDomain: true,
+        timeout: 3000,
+        async: false,
+        success: function (response) {
+            obj=[];
+            let keys = Object.keys(response.devices);
+            keys.forEach((i) => {
+              var myObj = {
```
Comment (akshatnitd) [inline src/components/MyDevices/MyDevices.react.js]: Use `let`

**Comment by Akshat-Jain** [comment]
@DNS-404 Done. Please review. Thanks :-)
mydevicesdns

**Comment by Akshat-Jain** [comment]
@akshatnitd Done. Please review. Thanks :-)
nodevices

**Comment by Akshat-Jain** [comment]
@anshumanv @imjog @PrP-11 I looked into the Codacy failure, but it looks like Codacy and Travis are in conflict with each other. If I fix Codacy, then Travis will fail. Not sure what to do.

**Comment by akshatnitd** [comment]
I think, let the mentors have the final call on this and let them merge

ᐧ

**Comment by mariobehling** [comment]
Thanks!

**Comment by mariobehling** [comment]
Please check travis.

**Comment by mariobehling** [comment]
One of the last changes resulted in the left sidebar not being aligned on top in the settings area. Please help to resolve this in following PRs as well. Thanks.

screenshot from 2018-05-28 01-01-25

**Pull request merged**